### PR TITLE
Wf api

### PR DESCRIPTION
**Pull request by tschelbs18**

### CODE REVIEW

File: src/habit_server/app.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    new_habit = request.json['habitname']
+    userhabit = UserHabit(username = cur_user.username, habitname = new_habit)
+    result = db_manager.delete_habit(userhabit)
+    if result.ok():
+        return "delete habit successful", 200
+    else:
+        return "cannot delete habit ", 404
+
+@app.route('/api/habits/logs', methods=['GET'])
+def get_activites():
```
Comment (DanielKoohmarey) [inline src/habit_server/app.py:60]: For this route, we decided we will remove start/end options. We will simply assume last 100 days, and assume this is returned correctly from the db manager

File: src/habit_server/app.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        # get activity with start and end day
+    else:
+        userhabit = UserHabit(username = cur_user.username, habitname = habitname)
+        return db_manager.get_activites(userhabit)
+
+@app.route('/api/habits/logs', methods=['POST'])
+def add_activites():
+    habitname = request.json['habitname']
+    timestamp = request.json['timestamp']
+    activity = UserActivity(username = cur_user.username, habitname = habitname, timestamp = timestamp)
```
Comment (DanielKoohmarey) [inline src/habit_server/app.py:81]: Here, you will want to convert the timestamp that is a string of form 'YYYY-MM-DD' to the python datetime object using your toDate function in the utils file, and pass that as the arg in the UserActivity

File: src/habit_server/db_manager.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
+"""Database manager for the habit server."""
```
Comment (DanielKoohmarey) [inline src/habit_server/db_manager.py:1]: Merge in Jared's changes from his branch after 11/30, assume his changes take priority in conflicts.

File: test/test_api.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            response1 = c.post(
+                '/api/habits',
+                data=json.dumps({'habitname': 'reading'}),
+                content_type='application/json',
+            )
+            assert response1.status_code == 200
+
+            # These 2 tests are failed becuase of 'SQLite DateTime type only accepts Python datetime and date objects as input'
+            # time= "2019-12-04"
+            # datetime = toDate(time)
```
Comment (DanielKoohmarey) [inline test/test_api.py:84]: the field we pass in datetime should be the YYYY-MM-DD format from time, no need to convert to different representation. Once you add changes to endpoint above this test should work.

File: src/habit_server/app.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    return db_manager.add_activity(activity)
+
+# render
+@app.route('/login', methods=['GET'])
```
Comment (jrdbuch) [inline src/habit_server/app.py:85]: handle POST as well, maybe @rzdao can integrate this endpoint in his login branch

**Pull request closed without merging**